docs/source/topics/data_types.rst

```diff
@@ -0,0 +1,508 @@
+* **String to pass to the** :py:func:`~aiida.plugins.factories.DataFactory`: ``array.kpoints``
+* **Aim**: store grids of k-points (in reciprocal space, for crystal structures), or
+  explicit list of k-points (optionally with a weight associated to each one). Can also
+  associate labels to (some of the) points, which is very useful for later plottings
+  band structures (and store them in ``BandsData`` objects).
+* **What is stored in the database**: like ``ArrayData``, the shape of the arrays and the name of the arrays
+* **What is stored in the file repository**:  like ``ArrayData``, the array data in numpy format
+* **Additional functionality**:
+
+.. _topics:data_types:materials:kpoints:automatic:
+
+Automatic computation of k-point paths
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+AiiDA provides a number of tools and wrappers to automatically compute k-point paths given a cell or a crystal structure.
+
+The main interface is provided by the two methods :py:func:`aiida.tools.data.array.kpoints.get_kpoints_path` and :py:func:`aiida.tools.data.array.kpoints.get_explicit_kpoints_path`.
+
+These methods are also conveniently exported directly as, e.g., ``aiida.tools.get_kpoints_path``.
+
+The difference between the two methods is the following:
+
+- :py:func:`~aiida.tools.data.array.kpoints.get_kpoints_path` returns a dictionary of k-point coordinates (e.g. ``{'GAMMA': [0. ,0. ,0. ], 'X': [0.5, 0., 0.], 'L': [0.5, 0.5, 0.5]}``, and then a list of tuples of endpoints of each segment, e.g. ``[('GAMMA', 'X'), ('X', 'L'), ('L', 'GAMMA')]`` for the :math:`\Gamma-X-L-\Gamma` path.
+- :py:func:`~aiida.tools.data.array.kpoints.get_explicit_kpoints_path`, instead, returns a list of kpoints that follow that path, with some predefined (but user-customizable) distance between points, e.g. something like ``[[0., 0., 0.], [0.05, 0., 0.], [0.1, 0., 0.], ...]``.
+
+Depending on how the underlying code works, one method might be preferred on the other.
+
+The docstring of the methods describes the expected parameters.
+The general interface requires always a ``StructureData`` as the first parameter ``structure``, as well as a string for the method to use (by default this is `seekpath <https://github.com/giovannipizzi/seekpath/>`_, but also the ``legacy`` method implemented in earlier versions of AiiDA is available; see description below).
+
+Additional parameters are passed as ``kwargs`` to the underlying implementation, that often accepts a different number of parameters.
+
+.. _topics:data_types:materials:kpoints:seekpath:
+
+Seekpath implementation
+^^^^^^^^^^^^^^^^^^^^^^^
+
+When specifying ``method='seekpath'``, the `seekpath <https://github.com/giovannipizzi/seekpath/>`_ library is used to generate the path.
+Note that this requires  ``seekpath`` to be installed (this is not available by default, in order to reduce the dependencies of AiiDA core, but can be easily installed using ``pip install seekpath``).
```

Comment:
Should we mention as the 'best' way to install this, to use instead the correct extras in `pip install aiida-core`?

Reply:
You mean to install the `atomic_tools` extras via:

```
 pip install aiida-core[atomic_tools]
```

?

Reply:
yes